ci/cirrus_general_ci.yml

```diff
@@ -38,6 +38,46 @@ linux_aarch64_test_task:
     python dev.py test
 
 
+musllinux_amd64_test_task:
+  container:
+    image: alpine
+    cpu: 8
+    memory: 32G
+
+  env:
+    PATH: $PWD:$PATH
+
+  pip_cache:
+    folder: ~/.cache/pip
+    populate_script: |
+      apk add openblas-dev python3 python3-dev openblas build-base gfortran git py3-pip
+      ln -sf $(which python3.10) python
+      pip install -vvv --upgrade numpy
+
+  setup_script: |
+    apk update
+    apk add openblas-dev python3 python3-dev openblas build-base gfortran git py3-pip
+    git submodule update --init
+
+    ln -sf $(which python3.10) python
+    # I don't think that the alpine py3-numpy package has the include dirs.
+    pip install -vvv --upgrade numpy
+
+  build_script: |
+    python -m pip install meson ninja cython pybind11 pythran cython
+    python -m pip install click rich_click doit pydevtool
+    python dev.py build
+
+  test_script: |
+    pip install pooch pytest
+    # split over several runs because there's a segfault in ndimage (gh17270)
+    python dev.py test -s _lib -s cluster -s constants
```

Comment:
This doesn't work, only the last submodule that's specified is picked up here. It should be one line per submodule.

Reply:
Question: did you check exit codes work, so test failures actually fail the job?

Reply:
No, good point

Reply:
> This doesn't work, only the last submodule that's specified is picked up here. It should be one line per submodule.

Is there a syntax to exclude one module, but do the others?

Reply:
Yes, that normally works.

Reply:
> Is there a syntax to exclude one module, but do the others?

Not in `dev.py` itself; it's not completely trivial to implement either that or making `-s` take multiple submodules. However, this may work, because we can pass any argument directly to `pytest`:
```
python dev.py test -- --ignore=scipy/ndimage/tests
```
I just did a quick test, and it does seem to work. It could be made even more granular, in case the segfault is just triggered from a single `test_*.py` file.

Reply:
What is the `--` for, that is just after test?

I think the exclude is important, otherwise the CI run will stop after the first module that has a failing test.

Reply:
Arguments after `--` are passed on to `pytest` directly (see `python dev.py test --help`)